<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>TFactSR: Enrichment Approach to Predict Which Transcription Factors Are Regulated • TFactSR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="TFactSR: Enrichment Approach to Predict Which Transcription Factors Are Regulated">
<meta property="og:description" content="TFactSR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TFactSR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/TFactSR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>TFactSR: Enrichment Approach to Predict Which
Transcription Factors Are Regulated</h1>
                        <h4 data-toc-skip class="author">Atsushi
Fukushima</h4>
            <address class="author_afil">
      RIKEN Center for Sustainable Resource
Science<br><a class="author_email" href="mailto:#"></a><a href="mailto:afukushima@gmail.com" class="email">afukushima@gmail.com</a>
      </address>
                  
      
      
      <div class="hidden name"><code>TFactSR.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Instructions for the TFactSR package.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><a href="http://www.tfacts.org/" class="external-link">TFactS</a> is to predict which are
the transcription factors (TFs), regulated in a biological condition
based on lists of differentially expressed genes (DEGs) obtained from
transcriptome experiments. This package is based on the TFactS concept
and expands it. It allows users to performe TFactS-like enrichment
approach. The package can import and use the original catalogue file
from the <a href="http://www.tfacts.org/" class="external-link">TFactS website</a> as well as
users’ defined catalogues of interest that are not supported by TFactS
(e.g., Arabidopsis).</p>
</div>
<div class="section level2">
<h2 id="methods---statistical-significance">Methods - Statistical significance<a class="anchor" aria-label="anchor" href="#methods---statistical-significance"></a>
</h2>
<p>This vignette is largely based on the <a href="http://www.tfacts.org/TFactS-new/TFactS-v2/index1.html" class="external-link">TFactS
manual</a>. For the details about TFactS, please also see <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkq149" class="external-link">the
original paper by Essaghir et al. (2010)</a>.</p>
<div class="section level3">
<h3 id="p-value">P-value<a class="anchor" aria-label="anchor" href="#p-value"></a>
</h3>
<p>Briefly, the current package assumes the Sign-Less catalogue, i.e. it
does not contain any regulation type information (up- or
down-regulation). TFactSR compares the list of query DEGs (up and/or
down) with a catalogue of target gene signatures. The core algorithm is
based on Fisher’s exact test using a contingency table as follows:</p>
<table class="table">
<thead><tr class="header">
<th>TF</th>
<th>DEGs: Present</th>
<th>DEGs: Absent</th>
<th>Total</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Catalogue: Present</td>
<td><em>k</em></td>
<td><em>m - k</em></td>
<td><em>m</em></td>
</tr>
<tr class="even">
<td>Catalogue: Absent</td>
<td><em>n - k</em></td>
<td><em>N + k - n - m</em></td>
<td><em>N - m</em></td>
</tr>
<tr class="odd">
<td>Total</td>
<td><em>n</em></td>
<td><em>N - n</em></td>
<td><em>N</em></td>
</tr>
</tbody>
</table>
<ul>
<li>
<span class="math inline">\(m\)</span> is the number of target genes
annotated for the TF under consideration</li>
<li>
<span class="math inline">\(n\)</span> is the number of query
genes</li>
<li>
<span class="math inline">\(N\)</span> is the number of regulations
in the catalog</li>
<li>
<span class="math inline">\(k\)</span> is the number of query genes
that are annotated as regulated by TF (i.e., the intersection between
the query and the TF signature)</li>
</ul>
<p><span class="math display">\[ Pval = \left(
            \begin{array}{c}
              m \\
              i
            \end{array}
          \right)
          \left(
            \begin{array}{c}
              N-m \\
              n-i
            \end{array}
          \right)
/
          \left(
          \begin{array}{c}
              N \\
              n
          \end{array}
          \right)
\]</span></p>
</div>
<div class="section level3">
<h3 id="e-value">E-value<a class="anchor" aria-label="anchor" href="#e-value"></a>
</h3>
<p>E-value is the number of tests done (<span class="math inline">\(T\)</span>) times the p-value.</p>
<p><span class="math inline">\(Eval = pval \times T\)</span></p>
</div>
<div class="section level3">
<h3 id="bh-corrected-p-value">BH-corrected P-value<a class="anchor" aria-label="anchor" href="#bh-corrected-p-value"></a>
</h3>
<p>Benjamini and Hochberg false discovery rate (FDR) controlling method:
this is based on Benjamini and Hochberg (1995) and is calculated using
p.adjust() function. Note that the current TFactSR package does not use
Q-value (Storey 2003) under default settings.</p>
</div>
<div class="section level3">
<h3 id="random-control-rc">Random Control (RC)<a class="anchor" aria-label="anchor" href="#random-control-rc"></a>
</h3>
<p>RC is the percentage of which a TF is called significant under a
certain E-value threshold after a random simulation of user lists in
specified number of repetitions:</p>
<p><span class="math display">\[ RD_{(TF)} = \frac{\#\left\{ Eval(TF)
\leq \lambda \right\} \times 100} { \#\left\{rep\right\} } \]</span></p>
</div>
</div>
<div class="section level2">
<h2 id="prerequisites">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h2>
<p>The TFactSR package requires (1) a list of DEGs and (2) a catalogue
of interest. For Arabidopsis, we prepared the catalogue based on <a href="http://arabidopsis.med.ohio-state.edu/" class="external-link">AtRegNet</a> and <a href="http://atrm.cbi.pku.edu.cn/" class="external-link">ATRM</a>. For human data, the package
can do the calculation using default settings.</p>
<p>The Supported organisms by the original TFactS are human, rat and
mouse genes. As you can see below, you can perform an enrichment
analysis which TFs are regulated if you have a list of DEGs and your
catalogue.</p>
</div>
<div class="section level2">
<h2 id="getting-started">Getting started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>For human/rat/mouse data, we can do the TFactS analysis as
follows.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TFactSR</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">DEGs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">catalog</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tftg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extractTFTG.html">extractTFTG</a></span><span class="op">(</span><span class="va">DEGs</span>, <span class="va">catalog</span><span class="op">)</span></span>
<span><span class="va">TFs</span> <span class="op">&lt;-</span> <span class="va">tftg</span><span class="op">$</span><span class="va">TFs</span></span>
<span><span class="va">all.targets</span> <span class="op">&lt;-</span> <span class="va">tftg</span><span class="op">$</span><span class="va">all.targets</span></span>
<span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateTFactS.html">calculateTFactS</a></span><span class="op">(</span><span class="va">DEGs</span>, <span class="va">catalog</span>, <span class="va">TFs</span>, <span class="va">all.targets</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      TFs   m  n    N k      p.value      e.value       FDR.BH RC</span></span>
<span><span class="co">## 8  FOXO3  78 18 6838 7 5.499085e-10 1.594735e-08 1.594735e-08  1</span></span>
<span><span class="co">## 7  FOXO1 161 18 6838 7 9.012711e-08 2.613686e-06 1.306843e-06  0</span></span>
<span><span class="co">## 9  FOXO4   9 18 6838 2 2.330683e-04 6.758980e-03 2.252993e-03  0</span></span>
<span><span class="co">## 12  IRF9   3 18 6838 1 7.877425e-03 2.284453e-01 5.711133e-02  0</span></span>
<span><span class="co">## 23 STAT1  61 18 6838 2 1.092615e-02 3.168585e-01 6.329948e-02  0</span></span>
<span><span class="co">## 19 SMAD5   5 18 6838 1 1.309644e-02 3.797969e-01 6.329948e-02  0</span></span></code></pre>
<p>Using the option “TF.col” and “TF.col”, we can specify the target
column of your catalogue dataset. Carefully you have to choose the
TF-target relationships as follows.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">AtCatalog</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GenesUp_SH1H</span><span class="op">)</span></span>
<span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extractTFTG.html">extractTFTG</a></span><span class="op">(</span><span class="va">GenesUp_SH1H</span>, <span class="va">AtCatalog</span>,</span>
<span>                     TF.col <span class="op">=</span> <span class="st">"TF"</span>,</span>
<span>                     TG.col <span class="op">=</span> <span class="st">"target.genes"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateTFactS.html">calculateTFactS</a></span><span class="op">(</span><span class="va">GenesUp_SH1H</span>, <span class="va">AtCatalog</span>, <span class="va">d</span><span class="op">$</span><span class="va">TFs</span>, <span class="va">d</span><span class="op">$</span><span class="va">all.targets</span>, TF.col <span class="op">=</span> <span class="st">"TF"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          TFs    m  n     N k    p.value    e.value    FDR.BH RC</span></span>
<span><span class="co">## 17 AT3G23250    3 74 18910 1 0.01169456  0.3742258 0.3720668  0</span></span>
<span><span class="co">## 15 AT2G47460    6 74 18910 1 0.02325417  0.7441336 0.3720668  0</span></span>
<span><span class="co">## 26 AT5G11260  280 74 18910 1 0.66914068 21.4125018 1.0000000  0</span></span>
<span><span class="co">## 1  AT1G04370    2 74 18910 0 1.00000000 32.0000000 1.0000000  0</span></span>
<span><span class="co">## 2  AT1G09530  649 74 18910 0 1.00000000 32.0000000 1.0000000  0</span></span>
<span><span class="co">## 3  AT1G24260 4101 74 18910 0 1.00000000 32.0000000 1.0000000  0</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Essaghir A, Toffalini F, Knoops L, Kallin A, van Helden J, Demoulin
JB: Transcription factor regulation can be accurately predicted from the
presence of target gene signatures in microarray gene expression data.
<a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkq149" class="external-link">Nucleic
Acids Res. 2010 Jun 1;38(11):e120</a>.</li>
<li>Essaghir A, Demoulin JB: A Minimal Connected Network of
Transcription Factors Regulated in Human Tumors and Its Application to
the Quest for Universal Cancer Biomarkers. Plos One 7 (6), 2012, <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0039666" class="external-link">e39666</a>.</li>
</ol>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> on the system on
which this document was compiled:</p>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Monterey 12.6.3</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Asia/Tokyo</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] TFactSR_0.0.0.9000 BiocStyle_2.28.0  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] vctrs_0.6.3         cli_3.6.1           knitr_1.43         </span></span>
<span><span class="co">##  [4] rlang_1.1.1         xfun_0.40           stringi_1.7.12     </span></span>
<span><span class="co">##  [7] purrr_1.0.2         textshaping_0.3.6   jsonlite_1.8.7     </span></span>
<span><span class="co">## [10] glue_1.6.2          rprojroot_2.0.3     htmltools_0.5.6    </span></span>
<span><span class="co">## [13] ragg_1.2.5          sass_0.4.7          rmarkdown_2.24     </span></span>
<span><span class="co">## [16] evaluate_0.21       jquerylib_0.1.4     fastmap_1.1.1      </span></span>
<span><span class="co">## [19] lifecycle_1.0.3     yaml_2.3.7          memoise_2.0.1      </span></span>
<span><span class="co">## [22] bookdown_0.35       BiocManager_1.30.22 stringr_1.5.0      </span></span>
<span><span class="co">## [25] compiler_4.3.1      fs_1.6.3            rstudioapi_0.15.0  </span></span>
<span><span class="co">## [28] systemfonts_1.0.4   digest_0.6.33       R6_2.5.1           </span></span>
<span><span class="co">## [31] magrittr_2.0.3      bslib_0.5.1         tools_4.3.1        </span></span>
<span><span class="co">## [34] pkgdown_2.0.7       cachem_1.0.8        desc_1.4.2</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Atsushi Fukushima.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
